kinto/plugins/accounts/authentication.py

```diff
@@ -1,10 +1,29 @@
 import bcrypt
 from pyramid import authentication as base_auth
 
+from kinto.core import utils
 from kinto.core.storage import exceptions as storage_exceptions
 
+from . import ACCOUNT_CACHE_KEY
+
 
 def account_check(username, password, request):
+    settings = request.registry.settings
+    hmac_secret = settings['userid_hmac_secret']
+    cache_key = utils.hmac_digest(hmac_secret, ACCOUNT_CACHE_KEY.format(username))
+    cache_ttl = settings.get('account_cache_ttl_seconds', 30)
+
+    # Check cache to see whether somebody has recently logged in with the same
+    # username and password.
+    cache = request.registry.cache
+    cache_result = cache.get(cache_key)
+
+    # Username and password is correct. No need to compare hashes
```

Comment:
I don't think `is correct` applies here. It would probably be `was already verified`